File: chapter_2/10_software_sales.py
"""
10. Software Sales
A software company sells a package that retails for $99. Quantity discounts are given according to the following table:
Quantity    Discount
10 - 19     20%
20 - 49     30%
50 - 99     40%
>= 100      50%
Design a Python program that asks the user to enter the number of packages purchased.
The program should then display the amount of the discount (if any) and the total amount of the purchase after the discount.
"""

quantity = 0
discount = 0.0
discount_amount = 0.0
amount_with_discount = 0.0
amount_without_discount = 0.0

error_occurred = False

quantity = int(input("Enter Quantity Of Packages Purchased: "))

if quantity >= 100:
    discount = 0.5
elif quantity >= 50:
    discount = 0.4
elif quantity >= 20:
    discount = 0.3
elif quantity >= 10:
    discount = 0.2
else:
    error_occurred = True

if not error_occurred:
    amount_without_discount = quantity * 99
    discount_amount = discount * amount_without_discount
    amount_with_discount = amount_without_discount - discount_amount

    print("Amount:", amount_without_discount)
    print("Discount:", discount * 100, "%")
    print("Discounted Amount:", amount_with_discount)
else:
    print("An error occurred! Please provide a valid number of packages purchased.")

File: exercises/C/chapter_2 - Conditionals/10_software_sales.c
// 10. Software Sales
// A software company sells a package that retails for $99. Quantity discounts are given according to the following table:
// Quantity    Discount
// 10 - 19     20%
// 20 - 49     30%
// 50 - 99     40%
// >= 100      50%
// Design a Python program that asks the user to enter the number of packages purchased.
// The program should then display the amount of the discount (if any) and the total amount of the purchase after the discount.
#include <stdio.h>
#include <stdbool.h>


int main(int argc, char const *argv[])
{
    int packages_purchase;
    double discount_percent;
    double unit_price;
    double total_amount;
    double discount_amount;
    double total_amount_after_discount;
    bool error_occurred;

    printf("Enter the number of packages purchased: ");
    scanf("%d", &packages_purchase);

    if (packages_purchase >= 10 && packages_purchase <= 19) 
        discount_percent = 0.2;
    else if (packages_purchase >= 20 && packages_purchase <= 49)
        discount_percent = 0.3;
    else if (packages_purchase >= 50 && packages_purchase <= 99)
        discount_percent = 0.4;
    else if (packages_purchase >= 100)
        discount_percent = 0.5;
    else
        error_occurred = true;
    
    if (!error_occurred)
    {
        unit_price = 99;
        total_amount = unit_price * packages_purchase;
        discount_amount = discount_percent * total_amount;
        total_amount_after_discount = total_amount - discount_amount;
        printf("Discount Percent: %.0lf%%\n", discount_percent * 100);
        printf("Discount Amount: %.2lf\n", discount_amount);
        printf("Total Amount: %.2lf\n", total_amount);
        printf("Total Amount After Discount: %.2lf\n", total_amount_after_discount);
    }
    else
        printf("Error: Invalid number of packages purchased.\n");


    return 0;
}
